### PR TITLE
Onboarding flow

### PR DESCRIPTION
**Pull request by itexpert120**
https://github.com/NEARBuilders/bos-keypom-webcomponent/assets/165388405/e673f9c1-9e92-4268-a628-d4b714e58a10

### CODE REVIEW

File: widget/Index.jsx [hunk 1]
```diff
@@ -1 +1,67 @@
...
+  },
+  blocks: {
+    Header: () => <Widget src="${config_account}/widget/components.Header" />,
+    Footer: () => <></>,
+  },
+  router: {
+    param: "page",
+    routes: {
+      home: {
+        path: "${config_account}/widget/page.home.Index",
```
Comment (elliotBraem) [inline widget/Index.jsx]: What is home used for?

Comment (elliotBraem) [reply widget/Index.jsx]: Can we remove if not used

File: widget/Index.jsx [hunk 2]
```diff
@@ -1 +1,67 @@
...
+      welcome: {
+        path: "${config_account}/widget/page.welcome.Index",
+        blockHeight: "final",
+        init: {
+          name: "Welcome",
+        },
+        default: true,
+        hide: true,
+      },
+      profileInfo: {
```
Comment (elliotBraem) [inline widget/Index.jsx]: Rename "profileInfo" to "onboarding"

Reflect that in page and components too

File: widget/page/profileInfo/Theme.jsx [hunk 3]
```diff
@@ -0,0 +1 @@
+return <div>Theme</div>;
```
Comment (elliotBraem) [inline widget/page/profileInfo/Theme.jsx]: What's left for Theme?

We can save this for a separate ticket, but please create the separate ticket then

File: widget/page/welcome/Index.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
+const { Button } = VM.require("${config_account}/widget/components.Button") || {
+  Button: () => <></>,
+};
+
+const { href } = VM.require("devs.near/widget/lib.url") || {
```
Comment (elliotBraem) [inline widget/page/welcome/Index.jsx:5]: Should we add aliases now? Or leave for a separate ticket

File: widget/components/profileInfo/CoverUploader.jsx [hunk 5]
```diff
@@ -0,0 +1,300 @@
+const { Image, Refresh } = VM.require("${config_account}/widget/icons") || {
```
Comment (elliotBraem) [inline widget/components/profileInfo/CoverUploader.jsx]: This is really nice component, although what's different between Avatar and Cover?

Can we make a generic "File Uploader" that takes "Preview: (img) => <>...</>" from props? 

Screenshot 2024-06-19 at 9 57 10 AM

File: widget/app/View.jsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
+const { Router } = VM.require("buildhub.near/widget/Router") || {
+  Router: () => <></>,
+};
+const { config, ...passProps } = props;
+if (!config) {
+  // TODO: get from settings (or default)
+  config = {
```
Comment (elliotBraem) [inline widget/app/View.jsx:7]: Delete this component and have it point to builddao.near/widget/app.view instead

File: widget/Index.jsx [hunk 7]
```diff
@@ -1 +1,67 @@
...
+        },
+        hide: true,
+      },
+    },
+  },
+};
+return (
+  <CSS>
+    <Widget
+      src="${config_account}/widget/app.View"
```
Comment (elliotBraem) [inline widget/Index.jsx:56]: builddao.near/widget/app.view

**Comment by elliotBraem** [comment]
When the pull request is ready, please include a video and mark it ready for review

File: widget/page/profileInfo/Theme.jsx (hunk 3 above)
Comment (itexpert120) [reply widget/page/profileInfo/Theme.jsx]: @saadiqbal-dev is going to work on it now

File: widget/page/welcome/Index.jsx (hunk 4 above)
Comment (itexpert120) [reply widget/page/welcome/Index.jsx:5]: We can create a seperete ticket

File: widget/components/profileInfo/CoverUploader.jsx (hunk 5 above)
Comment (itexpert120) [reply widget/components/profileInfo/CoverUploader.jsx]: Done!

File: widget/app/View.jsx (hunk 6 above)
Comment (itexpert120) [reply widget/app/View.jsx:7]: the builddao view page has black background, that's why I had to fork it,

Comment (elliotBraem) [reply widget/app/View.jsx:7]: Can you make a pull request to build dao one to remove black background being set here?

That can be moved to CSS

Then have this point to builddao.near/widget/app.view. We can merge it here, black for now

Take the video before black

**Pull request merged**